File: yht0827/1월/1월 17일/양궁 대회.java
import java.util.*;

class Solution {

	static int[] answer = new int[11];
	static int maxScore = Integer.MIN_VALUE;
	static int[] ryan;
	static int[] aPeach;
	static int N;

	public static int[] solution(int n, int[] info) {
		ryan = new int[11];
		aPeach = info.clone();
		N = n;
		dfs(0, 0);

		if (Arrays.stream(answer).sum() == 0) {
			return new int[] {-1};
		}

		return answer;
	}

	public static void dfs(int idx, int cnt) {
		if (cnt == N) {
			int ryanScore = 0, aPeachScore = 0;

			for (int i = 0; i < 11; i++) {
				if (aPeach[i] == 0 && ryan[i] == 0)
					continue;
				if (aPeach[i] < ryan[i])
					ryanScore += 10 - i;
				else
					aPeachScore += 10 - i;
			}

			if (ryanScore > aPeachScore) {
				int diff = ryanScore - aPeachScore;

				if (diff > maxScore) {
					maxScore = diff;
					answer = ryan.clone();
				} else if (diff == maxScore) {
					for (int i = 10; i >= 0; i--) {
						if (ryan[i] != answer[i]) {
							if (ryan[i] > answer[i]) {
								answer = ryan.clone();
							}
							return;
						}
					}
				}
			}

		} else {
			for (int i = idx; i < 11; i++) {
				ryan[i]++;
				dfs(i, cnt + 1);
				ryan[i]--;
			}
		}

	}
}

File: Eunleelee/2024-01/1월 17일/양궁대회.py
from itertools import product
def solution(n, info):
    info.reverse()
    ans = [-1]
    maxd = 0
    for wl in product((True, False), repeat=11):
        t = 0
        s = sum(info[i]+1 for i in range(11) if wl[i])
        if s <= n:
            apeach = sum(i for i in range(11) if not wl[i] and info[i])
            ryan = sum(i for i in range(11) if wl[i])
            d = ryan-apeach
            if d > maxd:
                maxd = d
                ans = [info[i]+1 if wl[i] else 0 for i in range(11)]
                ans[0] += n-s
    ans.reverse()
    return ans